lib/dns-manager.js

```diff
@@ -0,0 +1,318 @@
+    });
+}
+
+stopAndRemoveDnsServer = function (dnsId) {
+    return new Promise((fulfill, reject) => {
+        docker.listContainers({
+            all: true
+        }, (err, containers) => {
+
+            if (containers.some(c => c.Image === dnsId)) {
+                console.log("stopAndRemoveDnsServer: Found container with same image");
+                for (let containerInfo of containers) {
+                    if (containerInfo.Image === dnsId) {
+                        console.log("stopAndRemoveDnsServer: container with same image: " + containerInfo.Image);
+                        let container = docker.getContainer(containerInfo.Id);
+                        container.remove({force: true}, (err, data) => {
+                            if (err) {
+                                debug('[%s] Error removing Container: %O', dnsId, err);
+                                reject(err);
+                            } else
+                                debug('[%s] Removed container!', dnsId);
+                            let image = docker.getImage(dnsId);
+                            image.remove({force: true}, (err, data) => {
+                                if (err) {
+                                    debug('[%s] Error removing Image: %O', dnsId, err);
+                                    reject(err);
+                                } else {
+                                    debug('[%s] Removed image!', dnsId);
+                                    fulfill();
+                                }
+                            });
+                        });
+                    }
+                }
+            } else {
+                fulfill();
+            }
+        });
+    });
+}
+
+addDomainsToDns = function (filePath, domains) {
+    return new Promise(async (fulfill, reject) => {
+        let promiseArray = [];
+        for (let domain of domains) {
+            promiseArray.push(new Promise((fulfill2, reject2) => {
+                Domain.findOne({id: domain}, (err, dom) => {
+                    if (err) {
+                        debug('[%S] Error finding domain!', domain);
+                        reject2(err);
+                    } else if (!dom) {
+                        debug('[%S] Could not find domain!', domain);
+                        reject2("No Domain with this ID found");
+                    } else {
+                        fse.appendFileSync(filePath, "server=/");
+                        if (dom.subdomain) {
+                            fse.appendFileSync(filePath, dom.subdomain + ".");
+                        }
+                        if (dom.secondLevelDomain) {
+                            fse.appendFileSync(filePath, dom.secondLevelDomain + ".");
+                        }
+                        if (dom.topLevelDomain) {
+                            fse.appendFileSync(filePath, dom.topLevelDomain);
+                        }
+                        fse.appendFileSync(filePath, "/8.8.8.8\n");
```

Comment:
Not everyone might be happy to use the Google DNS - suggest to make this a configurable variable. University infrastructure provides might not even allowed to use the Google DNS (e.g., a uni hosting a journal).